pkg/yum/repository.go

```diff
@@ -114,7 +114,7 @@ func (r *Repository) Repomd() (*Repomd, int, error) {
 		return nil, 0, fmt.Errorf("error parsing Repomd URL: %w", err)
 	}
 	if resp, err = r.settings.Client.Get(repomdURL); err != nil {
-		return nil, resp.StatusCode, fmt.Errorf("GET error: %w", err)
+		return nil, erroredStatusCode(resp), fmt.Errorf("GET error: %w", err)
```

Comment:
if url is anything non-valid like "something/"  then the response object is nil, and calling  resp.StatusCode results in a panic